src/com/google/enterprise/adaptor/filenet/Params.java

```diff
@@ -0,0 +1,197 @@
+      Splitter.on(',').omitEmptyStrings().trimResults();
+
+  private final String contentEngineUrl;
+  private final String username;
+  private final String password;
+  private final String objectStore;
+
+  private final String objectFactory;
+  private final String displayUrl;
+  private final boolean isPublic;
+  private final boolean pushAcls;
+  private final String additionalWhereClause;
+  private final String deleteAdditionalWhereClause;
+  private final Set<String> includedMetadata;
+  private final Set<String> excludedMetadata;
+  private final String globalNamespace;
+  private final int maxFeedUrls;
+
+  public Params(Config config) throws InvalidConfigurationException {
+    contentEngineUrl = config.getValue("filenet.contentEngineUrl");
+    logger.log(Level.CONFIG, "filenet.contentEngineUrl: {0}", contentEngineUrl);
+    try {
+      new ValidatedUri(contentEngineUrl).logUnreachableHost();
+    } catch (URISyntaxException e) {
+      throw new InvalidConfigurationException(
+          "Invalid filenet.contentEngineUrl: " + e.getMessage());
+    }
+
+    objectStore = config.getValue("filenet.objectStore").trim();
+    if (objectStore.isEmpty()) {
+      throw new InvalidConfigurationException("filenet.objectStore may not be empty");
+    }
+    logger.log(Level.CONFIG, "filenet.objectStore: {0}", objectStore);
+
+    username = config.getValue("filenet.username");
+    password = config.getValue("filenet.password");
+
+    // TODO(bmj): load the object factor?
+    objectFactory = config.getValue("filenet.objectFactory").trim();
+    if (objectFactory.isEmpty()) {
+      throw new InvalidConfigurationException(
+          "filenet.objectFactory may not be empty");
+    }
+    logger.log(Level.CONFIG, "filenet.objectFactory: {0}", objectFactory);
+
+    // TODO(bmj): If empty, can I base it off of contentEngineUrl?
+    String workplaceUrl = config.getValue("filenet.displayUrl").trim();
+    if (workplaceUrl.endsWith("/getContent/")) {
+      workplaceUrl = workplaceUrl.substring(0, workplaceUrl.length() - 1);
+    }
+    if (workplaceUrl.contains("/getContent")
+        && workplaceUrl.endsWith("/getContent")) {
+      workplaceUrl += "?objectStoreName=" + objectStore
+          + "&objectType=document&versionStatus=1&vsId=";
+    } else {
+      workplaceUrl += "/getContent?objectStoreName=" + objectStore
+          + "&objectType=document&versionStatus=1&vsId=";
+    }
+    displayUrl = workplaceUrl;
+    logger.log(Level.CONFIG, "displayUrl: {0}", displayUrl);
+    try {
+      new ValidatedUri(displayUrl + "0").logUnreachableHost();
+    } catch (URISyntaxException e) {
+      throw new InvalidConfigurationException(
+          "Invalid displayUrl: " + e.getMessage());
+    }
+
+    isPublic = Boolean.parseBoolean(config.getValue("filenet.isPublic").trim());
+    logger.log(Level.CONFIG, "filenet.isPublic: " + isPublic);
+
+    String pushAclsStr = config.getValue("filenet.pushAcls").trim();
```

Comment:
I think this goes away, too. Do any of the v4 connectors making pushing ACLs optional?